cmd/krew/cmd/upgrade.go

```diff
@@ -53,21 +54,30 @@ kubectl krew upgrade foo bar"`,
 					return errors.Wrap(err, "failed to find all installed versions")
 				}
 				for _, receipt := range installed {
-					pluginNames = append(pluginNames, receipt.Name)
+					pluginNames = append(pluginNames, receipt.Status.Source.Name+"/"+receipt.Name)
 				}
 				ignoreUpgraded = true
 				skipErrors = true
 			} else {
 				// Upgrade certain plugins
-				pluginNames = args
+				for _, arg := range args {
+					if !validation.IsSafePluginName(arg) {
+						if isCanonicalName(arg) {
+							return errors.New("upgrade command does not support INDEX/PLUGIN syntax; just specify PLUGIN")
+						}
+						return unsafePluginNameErr(arg)
+					}
+					r, err := receipt.Load(paths.PluginInstallReceiptPath(arg))
+					if err != nil {
+						return errors.Wrapf(err, "receipt not found for %q", arg)
```

Comment:
nit: we don't know what the error is
```suggestion
						return errors.Wrapf(err, "read receipt %q", arg)
```